extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java

```diff
@@ -173,6 +172,11 @@ private Oauth2Configuration createConfig(ServiceExtensionContext context) {
                 .privateKeyAlias(privateKeyAlias)
                 .clientId(clientId)
                 .privateKeyResolver(privateKeyResolver)
-                .certificateResolver(certificateResolver).build();
+                .certificateResolver(certificateResolver)
+                .build();
+    }
+
+    private String mandatorySetting(ServiceExtensionContext context, String key) {
+        return requireNonNull(context.getSetting(key, null), format("Missing %s mandatory config: %s", name(), key));
```

Comment:
Seems weird to have the extension name in the middle of the error message, no?

Reply:
Could be good to know which extension requires that parameter, e.g. to have the possibility to check documentation

Reply:
that's fair! done